sympy/core/expr.py

```diff
@@ -792,7 +800,11 @@ def _eval_interval(self, x, a, b):
         else:
             B = self.subs(x, b)
             if B.has(S.NaN, S.Infinity, S.NegativeInfinity, S.ComplexInfinity):
-                B = limit(self, x, b)
+                if (b < a) != False:
```

Comment:
This should also be `a < b` (and then "-" in the limit call) to make sure that the limits are always taken on opposite sides.